assets/js/translate-analytics.js

```diff
@@ -0,0 +1,42 @@
+// Detects if the page is translated and notifies analyticis
+const getNewLanguage = () => {
+  const html = document.getElementsByTagName("html")[0];
+  return html.getAttribute("lang")?.valueOf();
+};
+
+// adapted from: https://analytical42.com/2022/detect-track-translations-ga4/
+export default () => {
+  // Start by checking if the MutationObserver API is available
+  if (typeof MutationObserver === "function") {
+    // Tell the observer to monitor for changes to HTML attributes
+    const config = { attributes: true };
+    // Build the function to run when a change is observed
+    const callback = mutationList => {
+      // Loop through each observed change
+      for (let i = 0; i < mutationList.length; i += 1) {
+        // Only do something if the change was on an attribute
+        if (mutationList[i].type === "attributes") {
+          if (mutationList[i].attributeName === "lang") {
+            // Send an event to the dataLayer
+            const newLanguage = getNewLanguage();
+            window.dataLayer = window.dataLayer || [];
+            window.dataLayer.push({
+              event: "translate"
+            });
+            if (newLanguage) {
+              window.dataLayer.push({
+                language: newLanguage
+              });
+            }
+          }
+        }
+      }
+    };
+    // Create the actual observer
+    const observer = new MutationObserver(callback);
+    // Attach the observer to the <title> tag
+    observer.observe(document.getElementsByTagName("title")[0], config);
```

Comment:
question: I wonder if this ends up firing on Turbolinks-enhanced navigations (I think the title changes there?). I'm concerned it might get noisy

Reply:
does `<title>` ever get a `lang` attribute? Should this get a different callback then the `<html>` one?

Reply:
The only browser that updates the title is Edge.  Technically it adds an attribute to every element it tries to translate.  It unfortunately does not update the lang in the HTML when it does this.  From what I can tell there is no way to get the translated language from the browser.  So in Edge we can detect a langage change event, just not what it becomes.

On the other hand Safari and Chrome both update the lang attribute on the html element.  The helper function above is what that use case is for.

As for turbo links this shouldn't log any of that noise.  As we filter for specific attribute mutations (lang and _msttext) before logging the event.

Reply:
I don't think you need to add the observer to the `<title>` tag.